packages/repository/src/relations/has-one/has-one.repository.ts

```diff
@@ -39,6 +40,14 @@ export interface HasOneRepository<Target extends Entity> {
     filter?: Pick<Filter<Target>, Exclude<keyof Filter<Target>, 'where'>>,
     options?: Options,
   ): Promise<Target>;
+
+  /**
+   * Delete multiple target model instances
```

Comment:
This does not make sense to me. In HasOne relation, there is always either no target model or one target model. I don't see why would the caller want to delete multiple instances or provide a filter?

IMO, the API should look like this:

```ts
delete(options?: Options): Promise<Count>;
```

I am also not sure whether to return `Count` (`0` = not found;  `1` = deleted), or whether to use `Boolean` (`false` = not found; `true` = deleted). I guess it's good to stay consistent with other `delete` methods and return `Count`.

Reply:
I agree hasOne is always either no target model or one target model. The problem is that actually there is no any check to avoid multiple targets as we had in the past. Adding multiple targets check was removed from the initial version that I tested a coupe of months ago. This is the reason why I kept the Count. Let me know if you want me to add the check that avoid to add multiple targets then I will return true or false as per your recommendation.
Do you agree with a separated example one-2-one application as I did?

Reply:
> I agree hasOne is always either no target model or one target model. The problem is that actually there is no any check to avoid multiple targets as we had in the past. Adding multiple targets check was removed from the initial version that I tested a coupe of months ago. This is the reason why I kept the Count.

I think we need to decouple the delete API (what parameters is the method accepting) from the value it returns.

Do we both agree that it does not make sense to provide a `filter` or `where` argument where deleting the other side of `hasOne` relation?

As for the return value, let's keep returning `Count` for consistency with other `delete` methods.